File: src/db_stats.py
#!/usr/bin/env python3
"""Helper script to print database stats as JSON."""
import json
import os
import sys

# Ensure src/ is on path for db import
sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
import db

db_path = sys.argv[1] if len(sys.argv) > 1 else db.DB_PATH

try:
    conn = db.get_connection(db_path)
    db.setup_schema(conn)
    total = conn.execute("SELECT COUNT(*) FROM skills").fetchone()[0]
    skills = conn.execute("SELECT COUNT(*) FROM skills WHERE type='skill'").fetchone()[0]
    refs = conn.execute("SELECT COUNT(*) FROM skills WHERE type='reference'").fetchone()[0]
    total_bytes = conn.execute("SELECT SUM(byte_size) FROM skills").fetchone()[0] or 0
    conn.close()
    print(json.dumps({"total": total, "skills": skills, "refs": refs, "bytes": total_bytes}))
except Exception as e:
    print(json.dumps({"error": str(e)}))
    sys.exit(1)

File: src/db_stats.js
#!/usr/bin/env node
/**
 * Helper script and in-process module to return database stats as JSON.
 * Replaces db_stats.py with pure Node.js better-sqlite3.
 */


const db = require('./db');

function getDbStats(dbPath = null) {
  const targetPath = dbPath || db.DB_PATH;
  const conn = db.getConnection(targetPath, false);
  try {
    db.setupSchema(conn);
    const totalRow = conn.prepare("SELECT COUNT(*) as cnt FROM skills").get();
    const skillsRow = conn.prepare("SELECT COUNT(*) as cnt FROM skills WHERE type='skill'").get();
    const refsRow = conn.prepare("SELECT COUNT(*) as cnt FROM skills WHERE type='reference'").get();
    const bytesRow = conn.prepare("SELECT SUM(byte_size) as b FROM skills").get();

    return {
      total: totalRow ? totalRow.cnt : 0,
      skills: skillsRow ? skillsRow.cnt : 0,
      refs: refsRow ? refsRow.cnt : 0,
      bytes: bytesRow && bytesRow.b ? bytesRow.b : 0
    };
  } finally {
    conn.close();
  }
}

if (require.main === module) {
  const customDbPath = process.argv[2] || null;
  try {
    const stats = getDbStats(customDbPath);
    console.log(JSON.stringify(stats));
  } catch (err) {
    console.log(JSON.stringify({ error: err.message }));
    process.exit(1);
  }
}

module.exports = {
  getDbStats
};
